npm/debugTools/dataplane/parser.go

```diff
@@ -0,0 +1,338 @@
+package dataplane
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+
+	"github.com/Azure/azure-container-networking/npm/util"
+)
+
+// Parser struct
+type Parser struct{}
+
+// ParseIptablesObject create a Go object from specified iptable by calling iptables-save within node
+func (p *Parser) ParseIptablesObject(tableName string) *Iptables {
+	iptableBuffer := bytes.NewBuffer(nil)
+	// TODO: need to get iptable's lock
+	cmdArgs := []string{"-t", string(tableName)}
```

Comment:
Good point lets add the locking mechanism for iptables here, 

```suggestion
	cmdArgs := []string{util.IptablesWaitFlag, entry.LockWaitTimeInSeconds, iptMgr.OperationFlag, entry.Chain,"-t", string(tableName)}
```

Reply:
I tried running the -w flag with the iptables-save command but it returns an invalid option error. Is there a function similar to GrabIptableLock I saw sometimes ago?

Reply:
https://github.com/Azure/azure-container-networking/blob/3bd237cbf11514c60656f82d21581a3cd47e6c5e/npm/iptm/iptm_linux.go#L10

Reply:
Now that I think about it, do we really need to get the iptables' lock here? Since I'm only doing a read operation and not modify anything.